### DISCUSSION TITLE
Behaviour of the producer internal queue when using same producer to write messages to multiple topics

### DISCUSSION THREAD

**Discussion by niro313A** [edenhill/librdkafka#3722] [Q&A] [answered]
Below librdkafka api function is used to produce messages to multiple topics (say single partition topics) using same RdKafka::Producer object with enabled idempotence. **Is the message ordering preserved across topics as internal queue of the producer is shared across all topics and partitions?** 

_ErrorCode produce (const std::string topic_name, int32_t partition, int msgflags, void *payload, size_t len, const void *key, size_t key_len, int64_t timestamp, RdKafka::Headers *headers, void *msg_opaque) ;_

Consider below scenario to illustrate the problem.

Produce api function is called for below messages.

1. Topic A, Global sequence number = 1
2. Topic A, Global sequence number = 2
3. Topic A, Global sequence number = 3
4. Topic B, Global sequence number = 4
5. Topic B, Global sequence number = 5
6. Topic B, Global sequence number = 6

**At any given time, is it possible the kafka cluster to contain messages in Topic B without some of messages in Topic A?
e.g. Topic B has Messages with Global sequence number = 4 and 5, But topic A doesn't have the Message with Global sequence number 3**

**Comment by edenhill** [comment, marked as answer]
https://github.com/edenhill/librdkafka/blob/master/INTRODUCTION.md#ordering-and-message-sequence-numbers
